module/cmd/peggy/cmd/eth_keys.go

```diff
@@ -0,0 +1,130 @@
+package cmd
+
+import (
+	"crypto/ecdsa"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/keys"
+	"github.com/ethereum/go-ethereum/accounts/keystore"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/spf13/cobra"
+	"github.com/tendermint/tendermint/libs/cli"
+)
+
+const flagPassphrase = "passphrase"
+
+// Commands registers a sub-tree of commands to interact with
+// local private key storage.
+func Commands(defaultNodeHome string) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "eth_keys",
+		Short: "Manage your application's ethereum keys",
+		Long: `Keyring management commands. Generated by the official Ethereum go library.
+
+The keyring supports the following backends:
+    test        Stores keys insecurely to disk. It does not prompt for a password to be unlocked
+                and it should be use only for testing purposes.
+`,
+	}
+
+	cmd.AddCommand(
+		AddKeyCommand(),
+	)
+
+	cmd.PersistentFlags().String(flags.FlagHome, defaultNodeHome, "The application home directory")
+	cmd.PersistentFlags().String(flags.FlagKeyringDir, "", "The client Keyring directory; if omitted, the default 'home' directory will be used")
+	cmd.PersistentFlags().String(flags.FlagKeyringBackend, flags.DefaultKeyringBackend, "Select keyring's backend (os|file|test)")
+	cmd.PersistentFlags().String(cli.OutputFlag, "text", "Output format (text|json)")
+
+	return cmd
+}
+
+// AddKeyCommand defines a keys command to generate a key
+func AddKeyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "add",
```

Comment:
Maybe we change the name here to `generate` as we have no way of listing the generated keys? Adding full CRUD to this command tree isn't necessary right now.

Reply:
Up to you, I was mostly trying to maintain symmetry with the `peggy keys add` command.

Reply:
This is fine we can merge as is